### PR TITLE
Important behaviour not described for `componentDidMount()`

### PR DESCRIPTION
**Pull request by TrySpace**

### CODE REVIEW

File: docs/docs/ref-03-component-specs.md [hunk 1]
```diff
@@ -126,6 +126,10 @@ If you want to integrate with other JavaScript frameworks, set timers using `set
 >
 > Prior to v0.9, the DOM node was passed in as the last argument. If you were using this, you can still access the DOM node by calling `this.getDOMNode()`.
 
+> Important:
+>
+> `componentDidMount()` will only fire on an element rendered with `React.render`, any children component will not trigger either mount lifecycles, `componentWillMount()`.
+
```
Comment (iamdustan) [inline docs/docs/ref-03-component-specs.md]: Is this note saying the following?

```javascript
var Parent = React.createClass({
  componentDidMount() { console.log('Parent componentDidMount called'); },
  render() {
    return <Child />;
  }
});

var Child = React.createClass({
  // the Important note in this PR is stating that this is not fired for this example.
  componentDidMount() { },
  render() {
    return <div>Child</div>;
  }
});

React.render(<App />, document.body);
```

Comment (iamdustan) [inline docs/docs/ref-03-component-specs.md]: because in reality all children `componentWillMount` and `componentDidMount` lifecycle methods *are* called.

Comment (zpao) [inline docs/docs/ref-03-component-specs.md]: Yea, this change looks wrong. What lead you to believe this?

**Comment by TrySpace** [comment]
@zpao This rather old reply to a question on SO: http://stackoverflow.com/questions/24832025/react-js-componentdidmount-not-called-when-using-rendercomponenttostring

 `Reacts componentDidMount method is only called after the initial render triggered by the React.renderComponent function. The Content component is never mounted using that method so the componentDidMount method is never called. I think what you're trying to do isn't possible using this method and there's probably a most appropriate method to achieve the result you are looking for.`

He sounded pretty sure of himself, maybe it has been this way?

And I was pretty sure that didmount fired on all children before, in many cases. But then I encountered a situation where no child element would fire.
In what rare case could it possibly not fire? 

Could it have to do with something `renderComponentToString` does? (which I don't use, at least not directly as far as I know)

I had a similar case as the above example, with just one more sub-child, which should not matter. And it just would not fire. The reason I needed it to have been mounted, is I'm generating children for https://github.com/alvarotrigo/fullPage.js dynamically, and initiate it when parent component has mounted, however. In random cases the parent's didMount method fired when not all children where there/mounted, at least according to fullPage, which just uses a jQuery selector. Thus I was looking for some way to check if they where, which brought me, eventually here,

(I eventually resorted to checking if any children with a specific class (gengerated by fullPage) are in the fullPage container, and only if it's `0` initiate fullPage, and checking this in parent's DidMount and each time in each DidUpdate cycle )

(I actually rather not use anything jQuery anymore since I found react, it's pretty messy, anyone know a (purejs) alternative to fullPage? other than onePage?)

**Comment by spicyj** [comment]
renderToString doesn't fully mount a component, so we don't call the componentDidMount method in that case (e.g., `this.getDOMNode()` wouldn't work because there isn't a DOM node at that point).

I'm going to close this out because I believe there hasn't been any behavior change here in React, but if you can post a minimal repro case where a component is mounted but `componentDidMount` isn't called, please post it here and we'll take a look.

**Pull request closed without merging**